File: taurus-protect-sdk-java/client/src/main/java/com/taurushq/sdk/protect/client/model/VisibilityGroup.java
package com.taurushq.sdk.protect.client.model;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * Represents a visibility group in the Taurus Protect system.
 * <p>
 * Visibility groups are used to control which users can see specific
 * wallets, addresses, and other entities. Users can only see entities
 * that belong to their visibility groups.
 *
 * @see VisibilityGroupUser
 * @see VisibilityGroupService
 */
public class VisibilityGroup {

    private String id;
    private String tenantId;
    private String name;
    private String description;
    private List<VisibilityGroupUser> users;
    private OffsetDateTime creationDate;
    private OffsetDateTime updateDate;
    private String userCount;

    public String getId() {
        return id;
    }

    public void setId(final String id) {
        this.id = id;
    }

    public String getTenantId() {
        return tenantId;
    }

    public void setTenantId(final String tenantId) {
        this.tenantId = tenantId;
    }

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(final String description) {
        this.description = description;
    }

    public List<VisibilityGroupUser> getUsers() {
        return users;
    }

    public void setUsers(final List<VisibilityGroupUser> users) {
        this.users = users;
    }

    public OffsetDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(final OffsetDateTime creationDate) {
        this.creationDate = creationDate;
    }

    public OffsetDateTime getUpdateDate() {
        return updateDate;
    }

    public void setUpdateDate(final OffsetDateTime updateDate) {
        this.updateDate = updateDate;
    }

    public String getUserCount() {
        return userCount;
    }

    public void setUserCount(final String userCount) {
        this.userCount = userCount;
    }
}

File: taurus-protect-sdk-go/pkg/protect/model/visibility_group.go
package model

import "time"

// VisibilityGroup represents a restricted visibility group in the system.
// Visibility groups control which users can access specific resources like wallets.
type VisibilityGroup struct {
	// ID is the unique identifier for the visibility group.
	ID string `json:"id"`
	// TenantID is the tenant this visibility group belongs to.
	TenantID string `json:"tenant_id,omitempty"`
	// Name is the display name of the visibility group.
	Name string `json:"name"`
	// Description is an optional description of the visibility group.
	Description string `json:"description,omitempty"`
	// Users is the list of users that belong to this visibility group.
	Users []*VisibilityGroupUser `json:"users,omitempty"`
	// UserCount is the number of users in this visibility group.
	UserCount int64 `json:"user_count"`
	// CreatedAt is when the visibility group was created.
	CreatedAt time.Time `json:"created_at"`
	// UpdatedAt is when the visibility group was last updated.
	UpdatedAt time.Time `json:"updated_at"`
}

// VisibilityGroupUser represents a user reference within a visibility group.
type VisibilityGroupUser struct {
	// ID is the unique identifier for the user.
	ID string `json:"id"`
	// ExternalUserID is the external identifier for the user.
	ExternalUserID string `json:"external_user_id,omitempty"`
}

// Note: User type is defined in user.go to avoid duplication
